flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActor.java

```diff
@@ -263,14 +274,41 @@ private void handleRpcInvocation(RpcInvocation rpcInvocation) {
 								if (throwable != null) {
 									promise.failure(throwable);
 								} else {
-									promise.success(value);
+									long resultSize = 0;
+									if (isRemoteSender) {
+										try {
+											resultSize = new SerializedValue(value).getByteArray().length;
+											if (resultSize > maximumFramesize) {
+												promise.failure(new Throwable("The result size exceeds the maximum " +
+													"size, result size is " + resultSize + ", maximum frame size is " +
+													maximumFramesize + " ."));
+											} else {
+												promise.success(value);
+											}
+										} catch (IOException e) {
+											promise.failure(e);
```

Comment:
Maybe wrap in `AkkaRpcException("Failed to serialize the result for RPC call %s.", e)`